plugins/kubernetes-backend/src/service/KubernetesBuilder.test.ts

```diff
@@ -205,6 +209,145 @@ describe('KubernetesBuilder', () => {
         ],
       });
     });
+
+    it('happy path: lists clusters with custom AuthStrategy and custom auth metadata', async () => {
+      objectsProviderMock = {
+        getKubernetesObjectsByEntity: jest
+          .fn()
+          .mockResolvedValue(happyK8SResult),
+        getCustomResourcesByEntity: jest.fn().mockImplementation(_ => {
+          return Promise.resolve(happyK8SResult);
+        }),
```

Comment:
Maybe you want to use this pattern everywhere?
```suggestion
        getCustomResourcesByEntity: jest.fn().mockResolvedValue(happyK8SResult),
```